### PR TITLE
Make it more strongly-typed when used with TypeScript

### LINKED ISSUE #24: TypeScript: Use `typed-query-selector` to detect element type in `delegateTarget`
https://github.com/g-plane/typed-query-selector

Also used in https://github.com/fregante/select-dom

Together with #19, most users will never need to use generics again 🥳

### PR DESCRIPTION
**Pull request by fregante**
Closes #24

### CODE REVIEW

**Comment by fregante** [comment]
@stof can you try this change if you have time? If you pass a tag name in the selector, the tag type will also be inferred

```ts
delegate(document, 'textarea.comment', 'keydown', event => {…});
// event will be of type `delegate.Event<HTMLTextAreaElement, KeyboardEvent>`
```

Use `npm install delegate-it@next`, it also includes your PR #19

File: index.ts [hunk 1]
```diff
@@ -82,7 +84,31 @@ function safeClosest(event: Event, selector: string): Element | void {
  * Delegates event to a selector.
  */
 function delegate<
-	TElement extends Element = Element,
+	Selector extends string,
+	TElement extends Element = ParseSelector<Selector, HTMLElement>,
+	TEventType extends EventType = EventType
+>(
+	base: EventTarget | Document | ArrayLike<Element> | string,
+	selector: string,
```
Comment (stof) [inline index.ts:92]: shouldn't this be `selector: Selector` ?

Comment (fregante) [reply index.ts]: Thanks for reviewing! That's right. I had forgotten to push this change

**Comment by stof** [comment]
@fregante I confirm that the 3.0.0-1 version has much better type inference than the 2.0.2 version when running on JS code

**Pull request merged**